### PR TITLE
Update notification architecture

### PR DESCRIPTION
**Pull request by mahinuzzaman**
Fixes #
Updated notification architecture 

- [x] Does not depend on other PRs
- [ ] Depends on other PR being merged first: #

## What kind of change does this PR introduce (check at least one)

- [ ] Bugfix
- [x] Improvement
- [ ] New feature
- [x] Refactor
- [ ] Infrastructure
- [ ] Other (please specify)

## How would you describe the changes in a changelog (1-2 lines)?

Provide a brief overview of the changes introduced in this pull request.

## Related Issues/PRs (optional)

Link to the relevant issue(s) or provide a clear description of the problem or feature being addressed.

## How to test

- [ ] Needs regression testing (add QA needed label)

## Checklist

- [x] I have tested these changes locally and they function as expected.
- [ ] I have added or updated necessary documentation to reflect the changes.
- [ ] I have followed the project's coding style and guidelines.
- [ ] I have added appropriate test coverage to ensure the changes are functioning correctly.
- [ ] I have checked for any potential conflicts with other branches or pull requests.
- [ ] I have updated necessary OpenAPI specs

## Reviewer Request (optional)

If you have any specific requests or suggestions for reviewers, mention them here.

## Screenshots (optional)

If applicable, provide any additional screenshots or GIFs that showcase the changes made in this pull request.

## Demo Link (optional)

If applicable, provide a link to a live demo or deployment where the changes can be reviewed.

## Known Issues (optional)

If there are any known issues or limitations with the changes made in this pull request, describe them here.

## Additional Notes (optional)

Add any additional notes or comments you think are relevant to this pull request.

## Deploy risk & follow up

Please assess risk & follow up on deployment of this PR. Elaborate your choice if selecting anything other than 'No risk'.

- [x] No risk
- [ ] Low
- [ ] High

### CODE REVIEW

File: core/application/bloodDonationWorkflow/BloodDonationService.ts [hunk 1]
```diff
@@ -195,39 +195,37 @@ export class BloodDonationService {
...
-        ...existingItem,
+      const updateData: Partial<DonorSearchDTO> = {
+        ...bloodDonationItem,
         id: requestPostId,
         retryCount: retryCount + 1
       }
 
       if (retryCount >= Number(process.env.MAX_RETRY_COUNT)) {
-        updateData.status = DonationStatus.EXPIRED
+        updateData.status = DonationStatus.COMPLETED
```
Comment (sakibstark11) [inline core/application/bloodDonationWorkflow/BloodDonationService.ts:227]: should this really be COMPLETED

Comment (mahinuzzaman) [reply core/application/bloodDonationWorkflow/BloodDonationService.ts:227]: this is not blood donation post entry. this the the status of donation search entry

Comment (sakibstark11) [reply core/application/bloodDonationWorkflow/BloodDonationService.ts:227]: 'The donor search process expired after the maximum retry limit is reached.' If the search process expired after trying, should it really be flagged as completed though?
It was not a successful request, COMPLETED tells me that it was successful

Comment (mahinuzzaman) [reply core/application/bloodDonationWorkflow/BloodDonationService.ts]: Updated in ef9d4eba2a85f2333cfc52a99408d24bd8c51ede

**Pull request merged**